### PR TITLE
Use Conan package manager to link jsmn

### LINKED ISSUE #69: Use Conan package manager to link jsmn

### PR DESCRIPTION
**Pull request by omdxp**
# Description

Resolves #69 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### CODE REVIEW

File: c/build.c [hunk 1]
```diff
@@ -241,7 +241,7 @@ void save_to_file(kuliya_schema *schema, const size_t slots_length, const char *
         kuliyas_with_terms[kuliya_with_terms_idx].__varname = malloc(varname_length + 1);
         memcpy(kuliyas_with_terms[kuliya_with_terms_idx].__varname, path_value, varname_length);
         kuliyas_with_terms[kuliya_with_terms_idx]
-            .__varname[varname_length + 1] = '\0';
+            .__varname[varname_length] = '\0';
```
Comment (omdxp) [inline c/build.c:244]: interesting this was not caught on macOS

File: c/helpers/string.h [hunk 2]
```diff
@@ -48,6 +48,15 @@ void remove_chars(char *str, int c, ...)
...
     va_end(args);
+
+    // This is to confirm deleting the first vararg character
+    size_t j = 0;
+    for (size_t i = 0; i < str[i] != '\0'; ++i)
+    {
+        if (str[i] != c)
+            str[j++] = str[i];
+    }
+    str[j] = '\0';
```
Comment (omdxp) [inline c/helpers/string.h:59]: another mystery between Linux and macOS 😅

**Review by ZibanPirate — CHANGES_REQUESTED** [review]
can you add a simple Readme:
- what needs to be installed: cmake, conan
- how to install conan packages
- how to run builds
- how to configure VSCode extensions

Because I could not make it run locally

**Comment by omdxp** [comment]
> can you add a simple Readme:
> 
> * what needs to be installed: cmake, conan
> * how to install conan packages
> * how to run builds
> * how to configure VSCode extensions
> 
> Because I could not make it run locally

Planning to add it on #76 once I finish this one

File: c/build.c (hunk 1 above)
Comment (ZibanPirate) [reply c/build.c:244]: hmm, could it be compiler related? are we using the same compilers in Linux and macOS?

File: c/helpers/string.h [hunk 3]
```diff
@@ -48,6 +48,15 @@ void remove_chars(char *str, int c, ...)
...
     va_end(args);
+
+    // This is to confirm deleting the first vararg character
+    size_t j = 0;
+    for (size_t i = 0; str[i] != '\0'; ++i)
+    {
+        if (str[i] != c)
+            str[j++] = str[i];
+    }
+    str[j] = '\0';
```
Comment (ZibanPirate) [inline c/helpers/string.h:59]: can we use [libunistring](https://conan.io/center/recipes/libunistring) instead?

**Review by ZibanPirate — APPROVED** [review]
🚀

File: c/helpers/string.h (hunk 3 above)
Comment (omdxp) [reply c/helpers/string.h:59]: seems a really good candidate to standardize text processing as long as this is a build dependency.

I will check it out and see what can I do with it.

Comment (omdxp) [reply c/helpers/string.h]: Using this library has solved these inconsistencies between strings on different platforms

File: c/build.c (hunk 1 above)
Comment (omdxp) [reply c/build.c:244]: you're right, I was using clang on my system, and the action was probably using gcc

**Pull request merged**